lib/contracts/Options/ERC721OptionContractTypeOne.js

```diff
@@ -103,6 +104,12 @@ async function settleERC721OptionNFT(
      * -> premiumTokenContract.connect(buyerAddress).approve(sellerAddress, premiumAmount);
      */
 
+    const buyerAccount = await getAccount(buyerAddress);
+    const premiumInDecimal = 10 ** Number(await premiumTokenContract.decimals()) * Number(premiumAmount);
+
+    console.log(`Authorizing premium payment of [${premiumAmount}, ${premiumInDecimal} in decimals] from Buyer [${buyerAddress}] to Seller [${sellerAddress}]`);
+    await premiumTokenContract.connect(buyerAccount).approve(erc721OptionNFTContract.address, premiumInDecimal); // Authorize payment to ERC721 contract.
+
```

Comment:
Lines 107 to 111 need to be in serverTest.js or in the front End - the WebServer will not have access to buyer account in the production version, only the manager account. The lines of code are correct .. they just need moving. It's only in this test end there is a backdoor to get to the buyer account. 

If the WebServer could get access to all user accounts it could steal all their assets .. buyer will login to MetaMask in the front end (or serverTest.js) and authorise the transfer .. the WebServer can then act on behalf of the buyer for the transfer .. but just for that authorised amount.